site/docs/getting-started/Querying.tsx

```diff
@@ -0,0 +1,115 @@
+import React from 'react'
+import CodeBlock from '@theme/CodeBlock'
+import Admonition from '@theme/Admonition'
+import Link from '@docusaurus/Link'
+import type { Dialect, PropsWithDialect } from './shared'
+
+const postgresqlCodeSnippet = `export async function insertPerson(person: Insertable<PersonTable>) {
+  return await db.insertInto('person')
+    .values(person)
+    .returningAll()
+    .executeTakeFirstOrThrow()
+}
+
+export async function deletePersonById(id: Person['id']) {
+  return await db.deleteFrom('person').where('id', '=', id)
+    .returningAll()
+    .executeTakeFirst()
+}`
+
+const dialectSpecificCodeSnippets: Record<Dialect, string> = {
+  postgresql: postgresqlCodeSnippet,
+  mysql: `export async function insertPerson(person: Insertable<PersonTable>) {
+  const { insertId } = await db.insertInto('person')
+    .values(person)
+    .executeTakeFirstOrThrow()
+      
+  return await findPersonById(insertId)
+}
+
+export async function deletePersonById(id: Person['id']) {
+  const person = await findPersonById(id)
+
+  if (person) {
+    await db.deleteFrom('person').where('id', '=', id).execute()
+  }
+
+  return person
+}`,
+  sqlite: postgresqlCodeSnippet,
+}
+
+export function Querying(props: PropsWithDialect) {
+  const dialect = props.dialect || 'postgresql'
+
+  const dialectSpecificCodeSnippet = dialectSpecificCodeSnippets[dialect]
+
+  return (
+    <>
+      <p>
+        <strong>Let's implement the person repository:</strong>
+      </p>
+      <CodeBlock language="ts" title="src/PersonRepository.ts" showLineNumbers>
+        {`import { Insertable, Selectable, Updateable } from 'kysely'
+import { db } from './database'
+import { PersonTable } from './types'
+
+export type Person = Selectable<PersonTable>
+
+export async function findPersonById(id: Person['id']) {
+  return await db.selectFrom('person')
+    .where('id', '=', id)
+    .selectAll()
+    .executeTakeFirst()
+}
+
+export async function findPeople(criteria: Partial<Person>) {

```

Comment:
Looking at this example, maybe I need to reconsider my opinion about the

```ts
where({ first_name: 'Sami' })
```

overload 🤔

Reply:
I had an `Object.entries` solution at first, but thought it might be too complex for some our readers. (not taking any assumptions on their JavaScript skills, only on them knowing SQL).